src/nidigital/system_tests/test_system_nidigital.py

```diff
@@ -36,3 +36,26 @@ def test_property_real64(multi_instrument_session):
 def test_property_string(multi_instrument_session):
     multi_instrument_session.start_label = 'foo'
     assert multi_instrument_session.start_label == 'foo'
+
+
+def test_tdr_all_channels(multi_instrument_session):
+    applied_offsets = multi_instrument_session.tdr(apply_offsets=False)
+    assert len(applied_offsets) == multi_instrument_session.channel_count
+
+    multi_instrument_session.apply_tdr_offsets(applied_offsets)
+
+    channels = [multi_instrument_session.get_channel_name(i) for i in
```

Comment:
We got this from IVI but still... how ANNOYING it is that this function is 1-based. What the hell were they thinking?

Reply:
sadness